assets/src/containers/AssignmentPlanningV2.js

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect } from 'react'
+import { withStyles } from '@material-ui/core/styles'
+import Paper from '@material-ui/core/Paper'
+import Grid from '@material-ui/core/Grid'
+import Spinner from '../components/Spinner'
+import ProgressBarV2 from '../components/ProgressBarV2'
+import AssignmentGradeBoxes from '../components/AssignmentGradeBoxes'
+import Error from './Error'
+import AssignmentTable from '../components/AssignmentTable'
+import Typography from '@material-ui/core/Typography'
+import { gql } from 'apollo-boost'
+import { useQuery } from '@apollo/react-hooks'
+import { calculateWeekOffset } from '../util/date'
+// import { DndProvider } from 'react-dnd'
+// import HTML5Backend from 'react-dnd-html5-backend'
+
+const styles = theme => ({
+  root: {
+    flexGrow: 1,
+    padding: 8
+  },
+  paper: {
+    padding: theme.spacing.unit * 2,
+    color: theme.palette.text.secondary
+  }
+})
+
+// const grades = {
+//   currentGrade: 85,
+//   goalGrade: null, // can be null
+//   maxPossibleGrade: 95,
+//   assignments: [
+//     {
+//       week: 1,
+//       dueDate: '10/15',
+//       title: 'Attendance',
+//       graded: true,
+//       score: 1,
+//       outOf: 1,
+//       percentOfFinalGrade: 5
+//     },
+//     {
+//       week: 1,
+//       dueDate: '10/15',
+//       title: 'Group Project',
+//       graded: true,
+//       score: 90,
+//       outOf: 100,
+//       percentOfFinalGrade: 15
+//     },
+//     {
+//       week: 2,
+//       dueDate: '10/22',
+//       title: 'Attendance',
+//       graded: false,
+//       score: null,
+//       outOf: 1,
+//       percentOfFinalGrade: 1
+//     },
+//     {
+//       week: 2,
+//       dueDate: '10/24',
+//       title: 'Discussion',
+//       graded: false,
+//       score: null,
+//       outOf: 5,
+//       percentOfFinalGrade: 20
+//     },
+//     {
+//       week: 3,
+//       dueDate: '11/24',
+//       title: 'Final Exam',
+//       graded: false,
+//       score: null,
+//       outOf: 100,
+//       percentOfFinalGrade: 50
+//     }
+//   ]
+// }
+
+function AssignmentPlanningV2 (props) {
+  const { classes, disabled, courseId } = props
+
+  const [assignments, setAssignments] = useState([])
+  const [goalGrade, setGoalGrade] = useState(null)
+
+  const setHandleAssignmentGoalGrade = (key, assignmentGoalGrade) => {
+    setAssignments([
+      ...assignments.slice(0, key),
+      { ...assignments[key], goalGrade: Number(assignmentGoalGrade) },
+      ...assignments.slice(key + 1)
+    ])
+  }
+
+  const { loading, error, data } = useQuery(gql`
+    {
+      course(courseId: 17700000000${courseId}) {
```

Comment:
@andrew-gardener any thoughts on the best way to not have to use the long Canvas course id here? Can the GQL endpoint be modified to accept the short course id?

Reply:
it could but honestly it might be good to be consistent with the graphql and use long ids for everything (since everything in canvas data is primary in long ids)

Reply:
Actually I'll add a canvasId filter (in addition to the courseId) so both are available

Reply:
Even the URLs for routing?